Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM openjdk:17-jdk-alpine@sha256:a996cdcc040704ec6badaf5fecf1e144c096e00231a29188596c784bcf858d05
```

Comment:
This is not quite the same as we use in our other Java apps (apart from the obvious Java 17 v Java 11 difference).

We currently use [Docker images provided by the Adoptium (formerly AdoptOpenJDK) project](https://hub.docker.com/u/adoptopenjdk) that use [Alpine Linux](https://www.alpinelinux.org/) as the operating system. Since Alpine works with [musl](https://musl.libc.org/) rather than [glibc](https://www.gnu.org/software/libc/), they toss in the [alpine-pkg-glibc](https://github.com/sgerrand/alpine-pkg-glibc) glibc compatibility layer and then include the standard glibc-based Linux build of the Eclipse Temurin (formerly AdoptOpenJDK) JRE.

I should note that while our Java apps have used this configuration for a long time, the Alpine Linux maintainers would disagree. They consider the glibc compatibility layer to be so dangerous that they’ve considered [making changes to Alpine Linux to stop it from working with alpine-pkg-glibc](https://github.com/docker-library/official-images/issues/10794). Their disapproval is having effects: [Adoptium will not be making Alpine images for Java 17 that use alpine-pkg-glibc](https://github.com/adoptium/containers/issues/1) and [there will not be official Alpine OpenJDK images that use alpine-pkg-glibc](https://github.com/docker-library/official-images/pull/10779) either.

What this PR uses is an [official OpenJDK Docker image](https://hub.docker.com/_/openjdk) that uses Alpine Linux as the operating system and a native musl-based Alpine Linux port of OpenJDK (developed under the name [Project Portola](https://openjdk.java.net/projects/portola/)) (not a JRE). The Alpine Linux OpenJDK port is not currently considered to be production ready.

That said, I think it’s fine to use for now. We can always re-evaluate later.

There’s also the JDK versus JRE difference. Since Java 11, the JRE has not really been a thing but the AdoptOpenJDK project continued to produce JRE binaries and images. Adoptium will only produce [Temurin JDK (and not JRE) builds for Java 17 and above](https://github.com/adoptium/temurin-build/issues/2683)) so at some point we’re going to have to deal with this change (and maybe look into the suggested alternative of using jlink).